### PR TITLE
[PIE-9133] Added note and points for branch selection logic

### PR DESCRIPTION
**Pull request by rashminandasahoo**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md [hunk 1]
```diff
@@ -220,7 +220,23 @@ The configurations of the required resources and any referenced entities like In
 
 #### Referenced entities are in the same repository
 
-If the referenced entities exist in the same repo, they are fetched from the same branch that you have selected for pipeline execution.​
+Here are the key points to keep in mind when the referenced entities reside in the same repository: 
+
+- During pipeline execution, Harness fetches the template from the same branch as the pipeline. This lets you test the templates before merging them back into the "default" branch.
+- Harness recommends creating a separate project for testing templates.
+- Templates and pipelines should be in the same repository.
```
Comment (mohitg0795) [inline docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md:227]: I think here we should mention "test pipelines". People shouldn't confuse it with the pipelines of their projects.

We can clearly say that Templates and the pipelines to test them should be in same repository.

File: docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md [hunk 2]
```diff
@@ -220,7 +220,23 @@ The configurations of the required resources and any referenced entities like In
...
+- Whenever you change a template in a feature branch, the corresponding testing pipelines are also updated.
```
Comment (mohitg0795) [inline docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md:228]: "the corresponding testing pipeline can also be updated if required". The pipelines aren't updated automatically.

File: docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md [hunk 3]
```diff
@@ -220,7 +220,23 @@ The configurations of the required resources and any referenced entities like In
...
+- As you can execute the pipeline from any branch, you can select the branch in which the changes to test templates were pushed. Merge the changes after testing. Upon merging, others will be able to access them.
+- This approach assumes the teams responsible for creating and managing templates will be different from those responsible for executing the pipeline (PE/DevOps teams versus Developers), so the test projects should be separate from the production projects.
+
+
+:::info
+Harness considers entities in the same repository if they are stored in different Git providers, but have the same repository names.
```
Comment (mohitg0795) [inline docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md:234]: Can we phrase it like this: "Harness considers entities to be in the same repository if the repository name matches for the entities, irrespective in whichever GIT provider the entities may exist."

File: docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md [hunk 4]
```diff
@@ -220,7 +220,23 @@ The configurations of the required resources and any referenced entities like In
...
+
+For example, when you store your pipeline in GitHub and the entities it references in Bitbucket, Harness treats the two repositories as one if their repository names are identical.
```
Comment (mohitg0795) [inline docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md:236]: Can we rephrase : "Harness treats the two repositories as "same" if their repository names are identical"

Comment (rashminandasahoo) [reply docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md]: Done.

File: docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md (hunk 3 above)
Comment (rashminandasahoo) [reply docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md]: Done

File: docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md (hunk 2 above)
Comment (rashminandasahoo) [reply docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md]: Done

File: docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md (hunk 1 above)
Comment (rashminandasahoo) [reply docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md]: Done.

File: docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md [hunk 5]
```diff
@@ -214,13 +214,33 @@ Click **Run**.
...
+- During pipeline execution, Harness fetches the entities like templates from the same branch as the pipeline. This lets you test the templates before merging them back into the "default" branch.
+- Harness recommends creating a separate project for testing templates.
+- Templates and the corresponding pipelines to test them should be in the same repository.
+- When you modify a template in a feature branch, the test pipeline is also present in the feature branch.
+- As you can execute the pipeline from any branch, you can select the branch in which the changes to test templates were pushed. Merge the changes after testing. Upon merging, others will be able to access them.
+- This approach works best when the teams responsible for creating and managing templates will be different from those responsible for executing the pipeline (Platform Engineering or DevOps teams versus Developers), so the test projects should be separate from the production projects.
+
+
+:::info
+Harness considers entities to be in the same repository if their repository names match, regardless of which GIT provider they use.
```
Comment (KapilDigani) [inline docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md:238]: This line is not required

File: docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md [hunk 6]
```diff
@@ -214,13 +214,33 @@ Click **Run**.
...
+
+For example, when you store your pipeline in GitHub and the entities it references in Bitbucket, Harness treats the two repositories as same if their repository names are identical.
```
Comment (KapilDigani) [inline docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md:240]: So is this

Comment (rashminandasahoo) [reply docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md:240]: @mohitg0795  Please see Kapil's comments.

**Review by mohitg0795 — DISMISSED** [review]

File: docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md [hunk 7]
```diff
@@ -214,13 +214,24 @@ Click **Run**.
 
 Click **Run Pipeline**.
 
-### Branch selection logic for referencing remote entities in pipelines
+### Branch selection logic for fetching referenced entities in remote pipelines
 
-The configurations of the required resources and any referenced entities like Input Sets, are fetched from Git during pipeline fetch, creation, or execution.
+The configurations of the required resources and any referenced entities like input sets, and templates are fetched from Git during pipeline fetch, creation, or execution.
```
Comment (brian-f-harness) [inline docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md:219]:
```suggestion
The configurations of the required resources and any referenced entities like input sets and templates are fetched from Git during pipeline fetch, creation, or execution.
```

File: docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md [hunk 8]
```diff
@@ -214,13 +214,24 @@ Click **Run**.
...
+- The referenced entities are stored in a different repository than the pipeline.
 
-If the referenced entities exist in the same repo, they are fetched from the same branch that you have selected for pipeline execution.​
+#### The referenced entities are stored in the same repository as the pipeline
+
+Following are the key points to keep in mind when the referenced entities reside in the same repository: 
+
+- During pipeline execution, Harness fetches the entities like templates from the same branch as the pipeline. This lets you test the templates before merging them back into the "default" branch.
+- Harness recommends creating a separate project for testing templates.
+- Templates and the corresponding pipelines to test them should be in the same repository.
```
Comment (brian-f-harness) [inline docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md:231]: Only if this is required, recommend updating to "must."

Comment (brian-f-harness) [inline docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md:231]:
```suggestion
- Templates and the corresponding pipelines to test them must be in the same repository.
```

File: docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md [hunk 9]
```diff
@@ -214,13 +214,24 @@ Click **Run**.
...
+- When you modify a template in a feature branch, the test pipeline is also present in the feature branch.
+- As you can execute the pipeline from any branch, you can select the branch in which the changes to test templates were pushed. Merge the changes after testing. Upon merging, others will be able to access them.
```
Comment (brian-f-harness) [inline docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md:233]:
```suggestion
- As you can execute the pipeline from any branch, you can select the branch in which the changes to test templates were pushed. Merge the changes after testing. Upon merging, others are able to access them.
```

File: docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md [hunk 10]
```diff
@@ -214,13 +214,24 @@ Click **Run**.
...
+- This approach works best when the teams responsible for creating and managing templates will be different from those responsible for executing the pipeline (Platform Engineering or DevOps teams versus Developers), so the test projects should be separate from the production projects.
```
Comment (brian-f-harness) [inline docs/platform/10_Git-Experience/configure-git-experience-for-harness-entities.md:234]:
```suggestion
- This approach works best when the teams responsible for creating and managing templates are different from those responsible for executing the pipeline (Platform Engineering or DevOps teams versus Developers), so the test projects should be separate from the production projects.
```

**Review by brian-f-harness — APPROVED** [review]
lgtm. Left a couple comments suggestions, not sure if they apply

**Pull request merged**